spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+    # TODO
+    
+    # step 6 : validate the potential merges with CC increase the contamination quality metrics
+    # TODO
+    
+    return potential_merges, corr_diff
+
+
+def compute_correlogram_diff(sorting, correlograms, bins,  correlogram_low_pass=800., adaptative_window_threshold=0.5,
+            pair_mask=None):
+    """
+    Original author: Aurelien Wyngaard ( lussac)
+    
+    Parameters
+    ----------
+    sorting: BaseSorting
+        The sorting object
+    correlograms: array 3d
+        The 3d array containing all cross and auto corrlogram
+    bins: array
+        Bins of the correlograms
+    correlogram_low_pass: float
+        low pass filter on correlogram for smoothing
+    adaptative_window_threshold
+        
+    pair_mask: None or boolean array
+        A bool matrix of size (num_units, num_units) to select
+        which pair to compute.
+    
+    Returns
+    -------
+    corr_diff
+    """
+    bin_ms = bins[1] - bins[0] 
+    
+    unit_ids = sorting.unit_ids
+    n = len(unit_ids)
+    
+    if pair_mask is None:
+        pair_mask = np.ones((n, n), dtype='bool')
+    
+    
+    # Index of the middle of the correlograms.
+    m = correlograms.shape[2] // 2
+    
+    num_spikes = sorting.get_total_num_spikes()
+    
+    # smooth correlogram by low pass filter
+    b, a = scipy.signal.butter(N=2, Wn = correlogram_low_pass / (1e3 / bin_ms /2), btype='low')
+    correlograms_smooth = scipy.signal.filtfilt(b, a, correlograms, axis=2)
+    
+    # find correlogram window for each units
+    win_sizes = np.zeros(n, dtype=int)
+    for unit_ind in range(n):
+        auto_corr = correlograms_smooth[unit_ind, unit_ind, :]
+        thresh = np.max(auto_corr) * adaptative_window_threshold
+        win_size = get_unit_adaptive_window(auto_corr, thresh)
+        win_sizes[unit_ind] = win_size
+        
+        # import matplotlib.pyplot as plt
+        # fig, ax = plt.subplots()
+        # bins2 = bins[:-1] + np.mean(np.diff(bins))
+        # ax.plot(bins2, auto_corr)
+        # ax.axhline(thresh)
+        # ax.axvline(bins2[m - win_size])
+        # ax.axvline(bins2[m + win_size])
+        # dev = -np.gradient(np.gradient(auto_corr))
+        # ax.plot(bins2, dev)
+        # plt.show()
+    
+    
+    
+    
+    
+    corr_diff = np.full((n, n), np.nan, dtype='float64')
+    for unit_ind1 in range(n):
+        for unit_ind2 in range(unit_ind1 + 1, n):
+            if not pair_mask[unit_ind1, unit_ind2]:
+                continue
+
+            unit_id1, unit_id2 = unit_ids[unit_ind1], unit_ids[unit_ind2]
+
+            
+            num1, num2 = num_spikes[unit_id1], num_spikes[unit_id2]
+            # Weighted window (larger unit imposes its window).
+            win_size = int(round((num1 * win_sizes[unit_ind1] + num2 * win_sizes[unit_ind2]) / (num1 + num2)))    
+            # Plage of indices where correlograms are inside the window.
+            corr_inds = np.arange(m - win_size, m + win_size, dtype=int)
```

Comment:
Now that I think about it, a `slice` object might be better for this.

Reply:
Yes but it will be harder to handle the shift. I keep this for now.